### ISSUE TITLE
Save Image As (Tweak)

### ISSUE THREAD

**Issue by KevinSanders** [choobin/cmis#24]
I've noticed when I want to use it, I'm already close to where I want to "save as" in the left tree. (ie: Animals > Cats)

Which in my case is "F:\Photo & Art Reference\Animals\Feline\Cats"

When we click the "Save Image As" button, It would be great if it automatically copies the "save path" of the selected item (as a starting point). Then the user could change the last item in the save path. In my case "Cats" to "Caracal".

**Comment by choobin** [comment]
Interesting. I will look into this. There are methods to set a start directory for nsIFilePicker. It might get a little hairy determining when it is appropriate to automatically add a path and when not to.

**Comment by KevinSanders** [comment]
I was thinking of that also, when the user opens the settings, nothing is selected on the left, so default behaviour would be to create it at the root level. But if the user is navigating through the list on the left, I'd imagine they were close to where they wanted it (that's just me thinking out loud of course). The main path (in my case) of "F:\Photo & Art Reference\" would be a nice starting point (as a default) when nothing is selected, as I'm sure most people have their particular place for storing images using this (GREAT!) add-on.

**Comment by choobin** [comment]
Firefox has a 'previous directory used/navigated to' variable that I could use. This might work here. In most cases the path will be close (or exactly) what is needed. 

By the way, just curious, are you experiencing any of the problems described in [issue/25](https://github.com/choobin/cmis/issues/25)? I am totally stumped. I have tested it on Windows/Linux/OSX with all FF versions from 9.0 to 18.0.1. I do not want to release all of these changes to AMO until I figure out what is going on.

**Comment by choobin** [comment]
Added in [75e5fbaa83](https://github.com/choobin/cmis/commit/75e5fbaa831652f556c65ca9002c616498e38b9b).

Each time the path selection dialog is used it will store the selected directory.

When a new item is created it will use the _root_ directory of the previously selected directory. I use the root directory here as I found that I was always moving back a directory during menu creation. If this becomes an issue I can always change it.

**Comment by KevinSanders** [comment]
"When a new item is created it will use the root directory of the previously selected directory." - Is it taking the path location from the item in the list?

I've installed the latest (20130128), If I create a new path item or when I select something in the list, and then add a new item. The path field is empty. I've previously just saved the zip for CMIS and that's where it's defaulting to (default firefox last save directory behaviour). But I'm not seeing any change from the previous version. :|

**Comment by choobin** [comment]
Ah. OK. I think I see what you mean. What I have implemented is as follows. 

When a new item is created and the path is selected it will store it. The next time an item is created _and_ the user clicks the select path button it will start from the previously stored root directory. This makes browsing for new items much easier as it will (in most cases) be in a very close position to the next required path.

Currently it does not add the stored path to newly created items (I think this is what you originally requested). I can just picture somebody submitting another issue stating that when they create a new item that they want to select the path for themselves. I do not know is this is just practical or a compromise. Although, I am happy to discuss it further.

**Comment by KevinSanders** [comment]
"When a new item is created and the path is selected it will store it. The next time an item is created and the user clicks the select path button it will start from the previously stored root directory. This makes browsing for new items much easier as it will (in most cases) be in a very close position to the next required path."
That sounds right. But when I do this, it doesn't seem to be going from the previous path/root. It still points to the last location I saved to using the standard firefox save (even though I've just created a new CMIS path since the default firefox save).

One other thing I was thinking, I assume we all use a particular parent folder for storing all of our images in various sub-folders. What about adding a settings item "Set Root"/ "Root Folder" (to select the default starting point) ? Then the user can clear it in their save path dialogue if they wanted to save somewhere else.

![root](https://f.cloud.github.com/assets/3275283/103822/9bee3aac-696e-11e2-829b-c6eb9895d233.png)
